src/highdicom/legacy/sop.py

```diff
@@ -25,384 +31,506 @@
+        return self._shared_tags[:]
+
+    @property
+    def SeriesInstanceUID(self) -> UID:
+        """Returns the sereis instance uid of the FrameSet
+
+        """
+        return self._frames[0].SeriesInstanceUID
+
+    @property
+    def StudyInstanceUID(self) -> UID:
+        """Returns the study instance uid of the FrameSet
+
+        """
+        return self._frames[0].StudyInstanceUID
+
+    def get_sop_instance_uid_list(self) -> list:
+        """Returns a list containing all SOPInstanceUID of the FrameSet
+
+        """
+        OutputList: list = []
+        for f in self._frames:
+            OutputList.append(f.SOPInstanceUID)
+        return OutputList
+
+    def get_sop_class_uid(self) -> UID:
+        """Returns the sop class uid of the FrameSet
+
+        """
+        return self._frames[0].SOPClassUID
+
+    def _find_per_frame_and_shared_tags(self) -> None:
+        """Detects and collects all shared and perframe attributes
+
+        """
+        rough_shared: dict = {}
+        sfs = self.frames
+        for ds in sfs:
+            for ttag, elem in ds.items():
+                if (not ttag.is_private and not
+                    ـDicomHelper.istag_file_meta_information_group(ttag) and not
+                        ـDicomHelper.istag_repeating_group(ttag) and not
+                        ـDicomHelper.istag_group_length(ttag) and not
+                        self._istag_excluded_from_perframe(ttag) and
+                        ttag != tag_for_keyword('PixelData')):
+                    elem = ds[ttag]
+                    if ttag not in self._perframe_tags:
+                        self._perframe_tags.append(ttag)
+                    if ttag in rough_shared:
+                        rough_shared[ttag].append(elem.value)
+                    else:
+                        rough_shared[ttag] = [elem.value]
+        to_be_removed_from_shared = []
+        for ttag, v in rough_shared.items():
+            v = rough_shared[ttag]
+            if len(v) < len(self.frames):
+                to_be_removed_from_shared.append(ttag)
+            else:
+                all_values_are_equal = True
+                for v_i in v:
+                    if not ـDicomHelper.isequal(v_i, v[0]):
+                        all_values_are_equal = False
+                        break
+                if not all_values_are_equal:
+                    to_be_removed_from_shared.append(ttag)
+        from pydicom.datadict import keyword_for_tag
+        for t, v in rough_shared.items():
+            if keyword_for_tag(t) != 'PatientSex':
+                continue
+        for t in to_be_removed_from_shared:
+            del rough_shared[t]
+        for t, v in rough_shared.items():
+            self._shared_tags.append(t)
+            self._perframe_tags.remove(t)
+
+    def _istag_excluded_from_perframe(self, t: BaseTag) -> bool:
+        return t in self.excluded_from_perframe_tags
+
+
+class FrameSetCollection:
```

Comment:
Please make this class "private", i.e., rename to `_FrameSetCollection`

Reply:
Same as [previeous](https://github.com/MGHComputationalPathology/highdicom/pull/34#discussion_r613715803).